src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -176,26 +183,42 @@ private void generateTypeScript() {
 
     new StyleFixPass(compiler, comments).process(externRoot, srcRoot);
 
+    new AstValidator(
+            compiler,
+            new AstValidator.ViolationHandler() {
+              @Override
+              public void handleViolation(String message, Node n) {
+                compiler.report(JSError.make(n, CLUTZ_MALFORMED_AST, message));
```

Comment:
Your rethrow below is actually stopping these errors from being printed at all -- if I comment it out, then I see the errors again. I don't know enough about the control flow here to make a suggestion though.

Your additional `.useSourceInfoFrom` calls seem to have added context to one of the three errors, but the other two are still showing up context-less. But I don't think that should block this PR.